NameLayerMain/src/vg/civcraft/mc/namelayer/group/Group.java

```diff
@@ -395,23 +396,38 @@ public static boolean areLinked(Group supergroup, Group subgroup) {
 		if (supergroup == null || subgroup == null) {
 			return false;
 		}
-		
-		return supergroup.hasSubGroup(subgroup) 
-				&& (subgroup.hasSuperGroup() 
-				&& subgroup.supergroup.equals(supergroup));
+		List <String> names = new LinkedList<String>();
+		names.add(supergroup.getName());
+		//check upwards to prevent cycles
+		if (supergroup.hasSuperGroup()) {
+			names.addAll(getRecursiveNamesUpwards(supergroup));
+		}
+		if (names.contains(subgroup.getName())) {
+			return true;
+		}
+		return false;
+	}
+	
+	private static List <String> getRecursiveNamesUpwards(Group g) {
```

Comment:
So funny story. 

If there is a cycle, this method will loop forever.

You have to add recursively and check at the same time. -- e.g. check if the name already exists before recursing. If it does exist, exit the check -- you know there is a loop. If it doesn't exist, keep going. A HashSet works better for that (quicker containment testing), and you should just do it in a loop instead of via recursion...